### PR TITLE
Replace `RunningApp.tag_to_object_id` with `function_ids` and `class_ids`

### PR DESCRIPTION
**Pull request by erikbern**
Slowly peeling apart all the old `indexed_objects` and `tag_to_object_id` nonsense

### CODE REVIEW

File: modal/runner.py [hunk 1]
```diff
@@ -163,7 +164,12 @@ async def _preload(tag, obj):
         async def _load(tag, obj):
             existing_object_id = tag_to_object_id.get(tag)
             await resolver.load(obj, existing_object_id)
-            running_app.tag_to_object_id[tag] = obj.object_id
+            if _Function._is_id_type(obj.object_id):
```
Comment (erikbern) [inline modal/runner.py:167]: this is a bit janky but i think we can revisit it later

File: modal/runner.py [hunk 2]
```diff
@@ -183,19 +189,16 @@ async def _publish_app(
     def filter_values(full_dict: dict[str, V], condition: Callable[[V], bool]) -> dict[str, V]:
```
Comment (erikbern) [inline modal/runner.py:189]: we can probably remove this helper since it's only used in one place now

**Comment by erikbern** [comment]
@prbot approve

**Review by mwaskom — COMMENTED** [review]
Yay! 🧹

**Pull request merged**